immutable_set/immutable_set.mbt

```diff
@@ -848,220 +833,172 @@ fn of_sorted_list[T : Compare](list : List[T]) -> ImmutableSet[T] {
-                height: 2,
-                value: 2,
-              },
-            ),
-            right: Node({ left: Empty, right: Empty, height: 1, value: 4 }),
-            height: 3,
-            value: 3,
-          },
-        ),
-        right: Node(
-          {
-            left: Node(
-              {
-                left: Node({ left: Empty, right: Empty, height: 1, value: 6 }),
-                right: Empty,
-                height: 2,
-                value: 7,
-              },
-            ),
-            right: Node({ left: Empty, right: Empty, height: 1, value: 9 }),
-            height: 3,
-            value: 8,
-          },
-        ),
-        height: 4,
-        value: 5,
-      },
-    ),
+    ImmutableSet::from_list(List::[7, 2, 9, 4, 5, 6, 3, 8, 1]).to_array(),
+    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].to_array(),
   )?
 }
 
 test "to_list" {
   @assertion.assert_eq(
-    ImmutableSet::from_list(List::[7, 2, 9, 4, 5, 6, 3, 8, 1]).to_list(),
-    List::[1, 2, 3, 4, 5, 6, 7, 8, 9],
+    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].to_array(),
+    [1, 2, 3, 4, 5, 6, 7, 8, 9],
+  )?
+}
+
+test "to_array" {
+  @assertion.assert_eq(
+    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].to_array(),
+    [1, 2, 3, 4, 5, 6, 7, 8, 9],
   )?
 }
 
 test "from_array" {
   @assertion.assert_eq(
-    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].to_list(),
-    List::[1, 2, 3, 4, 5, 6, 7, 8, 9],
+    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].to_array(),
+    [1, 2, 3, 4, 5, 6, 7, 8, 9],
   )?
 }
 
 test "remove_min" {
-  let set = ImmutableSet::from_list(List::[3, 4, 5]).remove_min()
-  @assertion.assert_eq(set.to_list(), List::[4, 5])?
+  @assertion.assert_eq(ImmutableSet::[3, 4, 5].remove_min().to_array(), [4, 5])?
 }
 
 test "add" {
   @assertion.assert_eq(
-    ImmutableSet::from_list(List::[7, 2, 9, 4, 6, 3, 8, 1]).add(5).to_list(),
-    List::[1, 2, 3, 4, 5, 6, 7, 8, 9],
+    ImmutableSet::[7, 2, 9, 4, 6, 3, 8, 1].add(5).to_array(),
+    [1, 2, 3, 4, 5, 6, 7, 8, 9],
   )?
+  @assertion.assert_eq(ImmutableSet::[2].add(1).to_array(), [1, 2])?
+  @assertion.assert_eq(ImmutableSet::[2].add(3).to_array(), [2, 3])?
+  @assertion.assert_eq(ImmutableSet::[2].add(1).add(3).to_array(), [1, 2, 3])?
 }
 
 test "remove" {
   @assertion.assert_eq(
-    ImmutableSet::from_list(List::[7, 2, 9, 4, 5, 6, 3, 8, 1]).remove(8).to_list(),
-    List::[1, 2, 3, 4, 5, 6, 7, 9],
+    ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].remove(8).to_array(),
+    [1, 2, 3, 4, 5, 6, 7, 9],
   )?
 }
 
 test "min" {
-  @assertion.assert_eq(
-    1,
-    ImmutableSet::from_list(List::[7, 2, 9, 4, 5, 6, 3, 8, 1]).min(),
+  @assertion.assert_eq(1, ImmutableSet::[7, 2, 9, 4, 5, 6, 3, 8, 1].min())?
+}
+
+test "is_empty" {
+  @assertion.assert_true(ImmutableSet::[].is_empty())?
+  @assertion.assert_false(ImmutableSet::[1].is_empty())?
+}
+
+test "to_string" {
+  inspect(
+    ImmutableSet::[1, 2, 3, 4, 5],
+    ~content="ImmutableSet::(1, 2, 3, 4, 5)",
```

Comment:
The compiler has given such a warning on these two lines, is this correct?
```
~\core\immutable_set\immutable_set.mbt:994:43-994:51 Warning 013: The type of this expression is (ImmutableSet[_/0]) -> Bool, which contains unresolved type variables. The type variable is default to Unit.
~\core\immutable_set\immutable_set.mbt:1003:11-1003:23 Warning 013: The type of this expression is (Array[_/0]) -> ImmutableSet[_/0], which contains unresolved type variables. The type variable is default to Unit.
```

Reply:
it is correct. `Set::[]` is too polymorphic that the compiler does not know its type.
cc @Guest0x0 , maybe we should relax the syntax to allow `Set[Int]::[]`?